octopusdeploy_framework/schemas/project.go

```diff
@@ -0,0 +1,379 @@
+						"url":                util.ResourceString().Required().Description("The URL associated with these version control settings.").Build(),
+						"username":           util.ResourceString().Required().Description("The username for the Git credential.").Build(),
+						"password":           util.ResourceString().Sensitive().Required().Description("The password for the Git credential").Build(), //util.GetPasswordResourceSchema(false),
+						"base_path":          util.ResourceString().Optional().Description("The base path associated with these version control settings.").Build(),
+						"default_branch":     util.ResourceString().Optional().Description("The default branch associated with these version control settings.").Build(),
+						"protected_branches": util.ResourceSet(types.StringType).Optional().Description("A list of protected branch patterns.").Build(),
+					},
+				},
+				Description: "Provides Git-related persistence settings for a version-controlled project.",
+			},
+			"jira_service_management_extension_settings": resourceSchema.ListNestedBlock{
+				NestedObject: resourceSchema.NestedBlockObject{
+					Attributes: map[string]resourceSchema.Attribute{
+						"connection_id":             util.ResourceString().Required().Description("The connection identifier associated with the extension settings.").Build(),
+						"is_enabled":                util.ResourceBool().Required().Description("Specifies whether or not this extension is enabled for this project.").Build(),
+						"service_desk_project_name": util.ResourceString().Required().Description("The project name associated with this extension.").Build(),
+					},
+				},
+				Description: "Provides extension settings for the Jira Service Management (JSM) integration for this project.",
+			},
+			"servicenow_extension_settings": resourceSchema.ListNestedBlock{
+				NestedObject: resourceSchema.NestedBlockObject{
+					Attributes: map[string]resourceSchema.Attribute{
+						"connection_id":                       util.ResourceString().Required().Description("The connection identifier associated with the extension settings.").Build(),
+						"is_enabled":                          util.ResourceBool().Required().Description("Specifies whether or not this extension is enabled for this project.").Build(),
+						"is_state_automatically_transitioned": util.ResourceBool().Required().Description("Specifies whether or not this extension will automatically transition the state of a deployment for this project.").Build(),
+						"standard_change_template_name":       util.ResourceString().Optional().Description("The name of the standard change template associated with this extension. If provided, deployments will create a standard change based on the provided template, otherwise a normal change will be created.").Build(),
+					},
+				},
+				Description: "Provides extension settings for the ServiceNow integration for this project.",
+			},
+			"template": resourceSchema.ListNestedBlock{
```

Comment:
Got some errors when setting up templates:
```
template {
    name = "template"
  }
```

>│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to octopusdeploy_project.project, provider "provider[\"octopus.com/com/octopusdeploy\"]" produced an unexpected new value: .template[0].label: was
│ null, but now cty.StringVal("").
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to octopusdeploy_project.project, provider "provider[\"octopus.com/com/octopusdeploy\"]" produced an unexpected new value:
│ .template[0].display_settings: was null, but now cty.MapValEmpty(cty.String).
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.